algorithms/kernel/dtrees/gbt/gbt_train_dense_default_impl.i

```diff
@@ -247,12 +247,28 @@ double TrainBatchTaskBase<algorithmFPType, BinIndexType, cpu>::computeLeafWeight
     algorithmFPType * pf      = f();
     val                       = -imp.g / val;
     const algorithmFPType inc = val * _par.shrinkage;
-
-    PRAGMA_IVDEP
-    PRAGMA_VECTOR_ALWAYS
-    for (size_t i = 0; i < n; ++i)
+    const size_t nThreads     = numAvailableThreads();
```

Comment:
Check this function - undefined behavior

Reply:
numAvailableThreads = maxThreads - n, where n is number of trees which are in building process at this moment